wallet/src/de/schildbach/wallet/ui/send/SendCoinsFragment.java

```diff
@@ -1021,24 +1029,26 @@ else if (everythingPlausible())
 
     private void handleGo() {
         privateKeyBadPasswordView.setVisibility(View.INVISIBLE);
+        attemptsRemainingTextView.setVisibility(View.GONE);
+        final String pin = privateKeyPasswordView.getText().toString().trim();
 
         if (wallet.isEncrypted()) {
             new DeriveKeyTask(backgroundHandler, application.scryptIterationsTarget()) {
                 @Override
                 protected void onSuccess(final KeyParameter encryptionKey, final boolean wasChanged) {
                     if (wasChanged)
                         application.backupWallet();
-                    signAndSendPayment(encryptionKey);
+                    signAndSendPayment(encryptionKey, pin);
                 }
-            }.deriveKey(wallet, privateKeyPasswordView.getText().toString().trim());
+            }.deriveKey(wallet, pin);
 
             setState(State.DECRYPTING);
         } else {
             signAndSendPayment(null);
```

Comment:
this needs to be changed based on the new method parameter `String pin` in `signAndSendPayment`.